File: naTanjir/naTanjir.Services/Database/Narudzba.cs
using System;
using System.Collections.Generic;

namespace naTanjir.Services.Database;

public partial class Narudzba:ISoftDelete
{
    public int NarudzbaId { get; set; }

    public int BrojNarudzbe { get; set; }

    public decimal UkupnaCijena { get; set; }

    public DateTime DatumKreiranja { get; set; }

    public DateTime? VrijemeBrisanja { get; set; }

    public string? StateMachine { get; set; }

    public bool IsDeleted { get; set; }

    public int KorisnikId { get; set; }

    public virtual Korisnici Korisnik { get; set; } = null!;

    public virtual ICollection<StavkeNarudzbe> StavkeNarudzbes { get; set; } = new List<StavkeNarudzbe>();
}

File: naTanjir/UI/natanjir_desktop/lib/models/narudzba.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:natanjir_desktop/models/korisnici.dart';
import 'package:natanjir_desktop/models/stavke_narudzbe.dart';
part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  int? narudzbaId;
  int? brojNarudzbe;
  double? ukupnaCijena;
  String? datumKreiranja;
  String? stateMachine;
  int? korisnikId;
  Korisnici? korisnik;
  List<StavkeNarudzbe>? stavkeNarudzbe;

  Narudzba();

  factory Narudzba.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
